Corona/Service/JHURepoDataService.swift

```diff
@@ -158,23 +159,21 @@ public class JHURepoDataService: DataService {
 
 			var series: [Date: Statistic] = [:]
 			for column in confirmedTimeSeries.values.indices {
-				let dateString = dateStrings[dateStrings.startIndex + column]
-				if let date = dateFormatter.date(from: dateString) {
-					var recoveredCount = 0
-					if let recoveredTimeSeries = recoveredTimeSeries {
-						recoveredCount = recoveredTimeSeries.values[min(column, recoveredTimeSeries.values.count - 1)]
-					}
-					var deathCount = 0
-					if let deathsTimeSeries = deathsTimeSeries {
-						deathCount = deathsTimeSeries.values[min(column, deathsTimeSeries.values.count - 1)]
-					}
-					let stat = Statistic(
-						confirmedCount: confirmedTimeSeries.values[column],
-						recoveredCount: recoveredCount,
-						deathCount: deathCount
-					)
-					series[date] = stat
+				let date = dateValues[column]
```

Comment:
In this case, if a date is not in the correct format, wouldn't this mean that we'll end up with wrong values? Maybe an assert that checks that `dateStrings.count == dateValues.count`?

Reply:
Yeah, that's a possibility, but not necessarily though. If a date can't be parsed you'd prefer ignoring a data column over having the app crash. Adding assert checks is risky here because we can't guarantee the validity of data we don't own or control.